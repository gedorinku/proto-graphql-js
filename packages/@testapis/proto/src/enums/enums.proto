syntax = "proto3";

package testapi.nested;

message MessageWithEnums {
  MyEnum1 my_enum_1 = 1;
}

enum MyEnum1 {
  MY_ENUM_1_UNSPECIRED = 0;
  FOO = 1;
  BAR = 2;
  BAZ = 3;
}
