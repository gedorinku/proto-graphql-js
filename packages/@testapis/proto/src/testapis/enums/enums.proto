syntax = "proto3";

package testapi.enums;

message MessageWithEnums {
  // Required.
  MyEnum required_my_enum = 1;
  // Optional.
  MyEnum optional_my_enum = 2;
  // Required.
  MyEnumWithoutUnspecified required_my_enum_without_unspecified = 3;
  // Optional.
  MyEnumWithoutUnspecified optional_my_enum_without_unspecified = 4;
}

enum MyEnum {
  MY_ENUM_UNSPECIFIED = 0;
  MY_ENUM_FOO = 1;
  // This is Bar.
  MY_ENUM_BAR = 2;
  MY_ENUM_BAZ = 3;
}


enum MyEnumWithoutUnspecified {
 MY_ENUM_WITHOUT_UNSPECIFIED_FOO = 0;
 MY_ENUM_WITHOUT_UNSPECIFIED_BAR = 1;
 MY_ENUM_WITHOUT_UNSPECIFIED_BAZ = 2;
}
