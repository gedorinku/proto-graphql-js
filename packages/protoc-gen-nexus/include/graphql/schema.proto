syntax = "proto3";

package graphql;

import "google/protobuf/descriptor.proto";

message GraphqlSchemaOptions {
  string type_prefix = 1;
}

extend google.protobuf.FileOptions {
  GraphqlSchemaOptions schema = 2056;
}

message GraphqlFieldOptions {
  bool ignore = 1;
}

extend google.protobuf.FieldOptions {
  GraphqlFieldOptions field = 2056;
}

message GraphqlOneofOptions {
  bool ignore = 1;
}

extend google.protobuf.OneofOptions {
  GraphqlOneofOptions oneof = 2056;
}

message GraphqlEnumValueOptions {
  bool ignore = 1;
}

extend google.protobuf.EnumValueOptions {
  GraphqlEnumValueOptions enum_value = 2056;
}
