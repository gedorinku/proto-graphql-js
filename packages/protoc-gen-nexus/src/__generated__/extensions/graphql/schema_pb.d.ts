// package: graphql
// file: graphql/schema.proto

import * as jspb from "google-protobuf";
import * as google_protobuf_descriptor_pb from "google-protobuf/google/protobuf/descriptor_pb";

export class GraphqlSchemaOptions extends jspb.Message {
  getTypePrefix(): string;
  setTypePrefix(value: string): void;

  getIgnoreRequests(): boolean;
  setIgnoreRequests(value: boolean): void;

  getIgnoreResponses(): boolean;
  setIgnoreResponses(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlSchemaOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlSchemaOptions): GraphqlSchemaOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlSchemaOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlSchemaOptions;
  static deserializeBinaryFromReader(message: GraphqlSchemaOptions, reader: jspb.BinaryReader): GraphqlSchemaOptions;
}

export namespace GraphqlSchemaOptions {
  export type AsObject = {
    typePrefix: string,
    ignoreRequests: boolean,
    ignoreResponses: boolean,
  }
}

export class GraphqlObjectTypeOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  getSquashUnion(): boolean;
  setSquashUnion(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlObjectTypeOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlObjectTypeOptions): GraphqlObjectTypeOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlObjectTypeOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlObjectTypeOptions;
  static deserializeBinaryFromReader(message: GraphqlObjectTypeOptions, reader: jspb.BinaryReader): GraphqlObjectTypeOptions;
}

export namespace GraphqlObjectTypeOptions {
  export type AsObject = {
    ignore: boolean,
    squashUnion: boolean,
  }
}

export class GraphqlInputObjectTypeOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlInputObjectTypeOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlInputObjectTypeOptions): GraphqlInputObjectTypeOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlInputObjectTypeOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlInputObjectTypeOptions;
  static deserializeBinaryFromReader(message: GraphqlInputObjectTypeOptions, reader: jspb.BinaryReader): GraphqlInputObjectTypeOptions;
}

export namespace GraphqlInputObjectTypeOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlFieldOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlFieldOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlFieldOptions): GraphqlFieldOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlFieldOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlFieldOptions;
  static deserializeBinaryFromReader(message: GraphqlFieldOptions, reader: jspb.BinaryReader): GraphqlFieldOptions;
}

export namespace GraphqlFieldOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlOneofOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlOneofOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlOneofOptions): GraphqlOneofOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlOneofOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlOneofOptions;
  static deserializeBinaryFromReader(message: GraphqlOneofOptions, reader: jspb.BinaryReader): GraphqlOneofOptions;
}

export namespace GraphqlOneofOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlEnumOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlEnumOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlEnumOptions): GraphqlEnumOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlEnumOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlEnumOptions;
  static deserializeBinaryFromReader(message: GraphqlEnumOptions, reader: jspb.BinaryReader): GraphqlEnumOptions;
}

export namespace GraphqlEnumOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

export class GraphqlEnumValueOptions extends jspb.Message {
  getIgnore(): boolean;
  setIgnore(value: boolean): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GraphqlEnumValueOptions.AsObject;
  static toObject(includeInstance: boolean, msg: GraphqlEnumValueOptions): GraphqlEnumValueOptions.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: GraphqlEnumValueOptions, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GraphqlEnumValueOptions;
  static deserializeBinaryFromReader(message: GraphqlEnumValueOptions, reader: jspb.BinaryReader): GraphqlEnumValueOptions;
}

export namespace GraphqlEnumValueOptions {
  export type AsObject = {
    ignore: boolean,
  }
}

  export const schema: jspb.ExtensionFieldInfo<GraphqlSchemaOptions>;

  export const objectType: jspb.ExtensionFieldInfo<GraphqlObjectTypeOptions>;

  export const inputType: jspb.ExtensionFieldInfo<GraphqlInputObjectTypeOptions>;

  export const field: jspb.ExtensionFieldInfo<GraphqlFieldOptions>;

  export const oneof: jspb.ExtensionFieldInfo<GraphqlOneofOptions>;

  export const enumType: jspb.ExtensionFieldInfo<GraphqlEnumOptions>;

  export const enumValue: jspb.ExtensionFieldInfo<GraphqlEnumValueOptions>;

