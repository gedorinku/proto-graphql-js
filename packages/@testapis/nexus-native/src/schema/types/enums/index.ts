export * from "./enums_pb_nexus";
