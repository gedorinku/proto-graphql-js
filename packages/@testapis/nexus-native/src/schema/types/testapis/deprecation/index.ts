export * from "./deprecation_pb_nexus";
export * from "./file_deprecation_pb_nexus";
