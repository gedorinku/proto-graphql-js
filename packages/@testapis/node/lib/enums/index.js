/*eslint-disable block-scoped-var, id-length, no-control-regex, no-magic-numbers, no-prototype-builtins, no-redeclare, no-shadow, no-var, sort-vars*/
"use strict";

var $protobuf = require("protobufjs/minimal");

// Common aliases
var $Reader = $protobuf.Reader, $Writer = $protobuf.Writer, $util = $protobuf.util;

// Exported root namespace
var $root = $protobuf.roots["default"] || ($protobuf.roots["default"] = {});

$root.testapi = (function() {

    /**
     * Namespace testapi.
     * @exports testapi
     * @namespace
     */
    var testapi = {};

    testapi.nested = (function() {

        /**
         * Namespace nested.
         * @memberof testapi
         * @namespace
         */
        var nested = {};

        nested.MessageWithEnums = (function() {

            /**
             * Properties of a MessageWithEnums.
             * @memberof testapi.nested
             * @interface IMessageWithEnums
             * @property {testapi.nested.MyEnum1|null} [myEnum_1] MessageWithEnums myEnum_1
             */

            /**
             * Constructs a new MessageWithEnums.
             * @memberof testapi.nested
             * @classdesc Represents a MessageWithEnums.
             * @implements IMessageWithEnums
             * @constructor
             * @param {testapi.nested.IMessageWithEnums=} [properties] Properties to set
             */
            function MessageWithEnums(properties) {
                if (properties)
                    for (var keys = Object.keys(properties), i = 0; i < keys.length; ++i)
                        if (properties[keys[i]] != null)
                            this[keys[i]] = properties[keys[i]];
            }

            /**
             * MessageWithEnums myEnum_1.
             * @member {testapi.nested.MyEnum1} myEnum_1
             * @memberof testapi.nested.MessageWithEnums
             * @instance
             */
            MessageWithEnums.prototype.myEnum_1 = 0;

            /**
             * Creates a new MessageWithEnums instance using the specified properties.
             * @function create
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {testapi.nested.IMessageWithEnums=} [properties] Properties to set
             * @returns {testapi.nested.MessageWithEnums} MessageWithEnums instance
             */
            MessageWithEnums.create = function create(properties) {
                return new MessageWithEnums(properties);
            };

            /**
             * Encodes the specified MessageWithEnums message. Does not implicitly {@link testapi.nested.MessageWithEnums.verify|verify} messages.
             * @function encode
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {testapi.nested.IMessageWithEnums} message MessageWithEnums message or plain object to encode
             * @param {$protobuf.Writer} [writer] Writer to encode to
             * @returns {$protobuf.Writer} Writer
             */
            MessageWithEnums.encode = function encode(message, writer) {
                if (!writer)
                    writer = $Writer.create();
                if (message.myEnum_1 != null && Object.hasOwnProperty.call(message, "myEnum_1"))
                    writer.uint32(/* id 1, wireType 0 =*/8).int32(message.myEnum_1);
                return writer;
            };

            /**
             * Encodes the specified MessageWithEnums message, length delimited. Does not implicitly {@link testapi.nested.MessageWithEnums.verify|verify} messages.
             * @function encodeDelimited
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {testapi.nested.IMessageWithEnums} message MessageWithEnums message or plain object to encode
             * @param {$protobuf.Writer} [writer] Writer to encode to
             * @returns {$protobuf.Writer} Writer
             */
            MessageWithEnums.encodeDelimited = function encodeDelimited(message, writer) {
                return this.encode(message, writer).ldelim();
            };

            /**
             * Decodes a MessageWithEnums message from the specified reader or buffer.
             * @function decode
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
             * @param {number} [length] Message length if known beforehand
             * @returns {testapi.nested.MessageWithEnums} MessageWithEnums
             * @throws {Error} If the payload is not a reader or valid buffer
             * @throws {$protobuf.util.ProtocolError} If required fields are missing
             */
            MessageWithEnums.decode = function decode(reader, length) {
                if (!(reader instanceof $Reader))
                    reader = $Reader.create(reader);
                var end = length === undefined ? reader.len : reader.pos + length, message = new $root.testapi.nested.MessageWithEnums();
                while (reader.pos < end) {
                    var tag = reader.uint32();
                    switch (tag >>> 3) {
                    case 1:
                        message.myEnum_1 = reader.int32();
                        break;
                    default:
                        reader.skipType(tag & 7);
                        break;
                    }
                }
                return message;
            };

            /**
             * Decodes a MessageWithEnums message from the specified reader or buffer, length delimited.
             * @function decodeDelimited
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {$protobuf.Reader|Uint8Array} reader Reader or buffer to decode from
             * @returns {testapi.nested.MessageWithEnums} MessageWithEnums
             * @throws {Error} If the payload is not a reader or valid buffer
             * @throws {$protobuf.util.ProtocolError} If required fields are missing
             */
            MessageWithEnums.decodeDelimited = function decodeDelimited(reader) {
                if (!(reader instanceof $Reader))
                    reader = new $Reader(reader);
                return this.decode(reader, reader.uint32());
            };

            /**
             * Verifies a MessageWithEnums message.
             * @function verify
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {Object.<string,*>} message Plain object to verify
             * @returns {string|null} `null` if valid, otherwise the reason why it is not
             */
            MessageWithEnums.verify = function verify(message) {
                if (typeof message !== "object" || message === null)
                    return "object expected";
                if (message.myEnum_1 != null && message.hasOwnProperty("myEnum_1"))
                    switch (message.myEnum_1) {
                    default:
                        return "myEnum_1: enum value expected";
                    case 0:
                    case 1:
                    case 2:
                    case 3:
                        break;
                    }
                return null;
            };

            /**
             * Creates a MessageWithEnums message from a plain object. Also converts values to their respective internal types.
             * @function fromObject
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {Object.<string,*>} object Plain object
             * @returns {testapi.nested.MessageWithEnums} MessageWithEnums
             */
            MessageWithEnums.fromObject = function fromObject(object) {
                if (object instanceof $root.testapi.nested.MessageWithEnums)
                    return object;
                var message = new $root.testapi.nested.MessageWithEnums();
                switch (object.myEnum_1) {
                case "MY_ENUM_1_UNSPECIRED":
                case 0:
                    message.myEnum_1 = 0;
                    break;
                case "FOO":
                case 1:
                    message.myEnum_1 = 1;
                    break;
                case "BAR":
                case 2:
                    message.myEnum_1 = 2;
                    break;
                case "BAZ":
                case 3:
                    message.myEnum_1 = 3;
                    break;
                }
                return message;
            };

            /**
             * Creates a plain object from a MessageWithEnums message. Also converts values to other types if specified.
             * @function toObject
             * @memberof testapi.nested.MessageWithEnums
             * @static
             * @param {testapi.nested.MessageWithEnums} message MessageWithEnums
             * @param {$protobuf.IConversionOptions} [options] Conversion options
             * @returns {Object.<string,*>} Plain object
             */
            MessageWithEnums.toObject = function toObject(message, options) {
                if (!options)
                    options = {};
                var object = {};
                if (options.defaults)
                    object.myEnum_1 = options.enums === String ? "MY_ENUM_1_UNSPECIRED" : 0;
                if (message.myEnum_1 != null && message.hasOwnProperty("myEnum_1"))
                    object.myEnum_1 = options.enums === String ? $root.testapi.nested.MyEnum1[message.myEnum_1] : message.myEnum_1;
                return object;
            };

            /**
             * Converts this MessageWithEnums to JSON.
             * @function toJSON
             * @memberof testapi.nested.MessageWithEnums
             * @instance
             * @returns {Object.<string,*>} JSON object
             */
            MessageWithEnums.prototype.toJSON = function toJSON() {
                return this.constructor.toObject(this, $protobuf.util.toJSONOptions);
            };

            return MessageWithEnums;
        })();

        /**
         * MyEnum1 enum.
         * @name testapi.nested.MyEnum1
         * @enum {number}
         * @property {number} MY_ENUM_1_UNSPECIRED=0 MY_ENUM_1_UNSPECIRED value
         * @property {number} FOO=1 FOO value
         * @property {number} BAR=2 BAR value
         * @property {number} BAZ=3 BAZ value
         */
        nested.MyEnum1 = (function() {
            var valuesById = {}, values = Object.create(valuesById);
            values[valuesById[0] = "MY_ENUM_1_UNSPECIRED"] = 0;
            values[valuesById[1] = "FOO"] = 1;
            values[valuesById[2] = "BAR"] = 2;
            values[valuesById[3] = "BAZ"] = 3;
            return values;
        })();

        return nested;
    })();

    return testapi;
})();

module.exports = $root;
