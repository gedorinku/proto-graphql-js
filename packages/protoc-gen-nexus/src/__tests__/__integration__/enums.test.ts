import { queryField } from "nexus";
import * as pbjs from "@testapis/node/lib/testapis/enums";
// import * as pbnative from "@testapis/node-native/lib/testapis/wktypes/wktypes_pb";
import { graphql } from "graphql";
import { testSchemaGeneration } from "../__helpers__/process.test.helper";

testSchemaGeneration("enums", "protobufjs", {
  schemaTests: [
    [
      "query",
      {
        test: queryField("test", {
          type: "MessageWithEnums",
          resolve() {
            return new pbjs.testapi.enums.MessageWithEnums({
              requiredMyEnum: pbjs.testapi.enums.MyEnum.MY_ENUM_BAR,
              requiredMyEnumWithoutUnspecified:
                pbjs.testapi.enums.MyEnumWithoutUnspecified.MY_ENUM_WITHOUT_UNSPECIFIED_FOO,
            });
          },
        }),
      },
      async (schema) => {
        expect(
          await graphql(
            schema,
            `
              query Test {
                test {
                  ...Message
                }
              }
              fragment Message on MessageWithEnums {
                requiredMyEnum
                requiredMyEnumWithoutUnspecified
                optionalMyEnum
                optionalMyEnumWithoutUnspecified
              }
            `
          )
        ).toMatchSnapshot();
      },
    ],
  ],
});

testSchemaGeneration("enums", "native protobuf");
