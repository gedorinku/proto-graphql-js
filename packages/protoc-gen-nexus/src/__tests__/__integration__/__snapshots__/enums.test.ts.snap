// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema generation with native protobuf can make GraphQL schema from generated DSLs: schema.graphql 1`] = `
"### This file was generated by Nexus Schema
### Do not make changes to this file directly


type MessageWithEnums {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnum: MyEnum

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnums: [MyEnum!]

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecified: MyEnumWithoutUnspecified

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnum: MyEnum!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnums: [MyEnum!]!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecified: MyEnumWithoutUnspecified!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]!
}

input MessageWithEnumsInput {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnum: MyEnum

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnums: [MyEnum!]

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecified: MyEnumWithoutUnspecified

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnum: MyEnum!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnums: [MyEnum!]!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecified: MyEnumWithoutUnspecified!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]!
}

enum MyEnum {
  \\"\\"\\"This is Bar.\\"\\"\\"
  MY_ENUM_BAR
  MY_ENUM_BAZ
  MY_ENUM_FOO
}

enum MyEnumWithoutUnspecified {
  MY_ENUM_WITHOUT_UNSPECIFIED_BAR
  MY_ENUM_WITHOUT_UNSPECIFIED_BAZ
  MY_ENUM_WITHOUT_UNSPECIFIED_FOO
}

type Query {
  ok: Boolean!
}
"
`;

exports[`schema generation with protobufjs can make GraphQL schema from generated DSLs: schema.graphql 1`] = `
"### This file was generated by Nexus Schema
### Do not make changes to this file directly


type MessageWithEnums {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnum: MyEnum

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnums: [MyEnum!]

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecified: MyEnumWithoutUnspecified

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnum: MyEnum!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnums: [MyEnum!]!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecified: MyEnumWithoutUnspecified!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]!
}

input MessageWithEnumsInput {
  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnum: MyEnum

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnums: [MyEnum!]

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecified: MyEnumWithoutUnspecified

  \\"\\"\\"Optional.\\"\\"\\"
  optionalMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnum: MyEnum!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnums: [MyEnum!]!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecified: MyEnumWithoutUnspecified!

  \\"\\"\\"Required.\\"\\"\\"
  requiredMyEnumWithoutUnspecifieds: [MyEnumWithoutUnspecified!]!
}

enum MyEnum {
  \\"\\"\\"This is Bar.\\"\\"\\"
  MY_ENUM_BAR
  MY_ENUM_BAZ
  MY_ENUM_FOO
}

enum MyEnumWithoutUnspecified {
  MY_ENUM_WITHOUT_UNSPECIFIED_BAR
  MY_ENUM_WITHOUT_UNSPECIFIED_BAZ
  MY_ENUM_WITHOUT_UNSPECIFIED_FOO
}

type Query {
  ok: Boolean!
}
"
`;

exports[`schema generation with protobufjs query 1`] = `
Object {
  "data": Object {
    "test": Object {
      "optionalMyEnum": null,
      "optionalMyEnumWithoutUnspecified": "MY_ENUM_WITHOUT_UNSPECIFIED_FOO",
      "requiredMyEnum": "MY_ENUM_BAR",
      "requiredMyEnumWithoutUnspecified": "MY_ENUM_WITHOUT_UNSPECIFIED_FOO",
    },
  },
}
`;
